src/core/reducer.ts

```diff
@@ -118,8 +118,18 @@ function initializeDeltalog(
     logEntry.metadata = pluginLogMetadata;
   }
 
-  if (typeof move === 'object' && move.redact === true) {
-    logEntry.redact = true;
+  function IsRedactFunction(
+    redact: boolean | ((...args: any[]) => any)
+  ): redact is (...args: any[]) => any {
+    return redact instanceof Function;
+  }
+
+  if (typeof move === 'object') {
+    if (IsRedactFunction(move.redact)) {
```

Comment:
I think this can be easier to read with less nesting:
```
  if (typeof move === 'object' && move.redact === true) {
    logEntry.redact = true;
  else if (typeof move === 'object' && move.redact instanceof Function) {
   logEntry.redact = move.redact(state.G, state.ctx);
  }

Reply:
I went ahead and applied this change to enable us to merge.